src/launchpad/size/treemap/macho_element_builder.py

```diff
@@ -59,220 +57,372 @@ def build_element(self, file_info: FileInfo, display_name: str) -> TreemapElemen
+                        path=None,
+                        is_dir=False,
+                        children=[
+                            TreemapElement(
+                                name=s.mangled_name,
+                                size=s.size,
+                                type=TreemapType.MODULES,
+                                path=None,
+                                is_dir=False,
+                                children=[],
+                            )
+                            for s in other_syms[:50]
+                        ],
+                    )
+                )
 
-        # ------------------------------------------------------------------ #
-        # 4.  Raw segments/sections (minus whatever the symbols already took) #
-        # ------------------------------------------------------------------ #
-        for segment in segments:
+        if other_symbols_children:
+            binary_children.append(
+                TreemapElement(
+                    name="Other Symbols",
+                    size=total_other_symbols_size,
+                    type=TreemapType.MODULES,
+                    path=None,
+                    is_dir=False,
+                    children=other_symbols_children,
+                )
+            )
+
+    def _add_segments(
+        self,
+        binary_analysis: MachOBinaryAnalysis,
+        binary_children: List[TreemapElement],
+        section_subtractions: Dict[str, int],
+    ) -> None:
+        for segment in binary_analysis.segments:
             segment_name = segment.name
-            segment_size = segment.size
             segment_children: List[TreemapElement] = []
 
-            segment_symbol_bytes = segment_subtractions.get(segment_name, 0)
-
             if segment.sections:
                 for section in segment.sections:
                     section_name = section.name
                     section_size = section.size
-
                     if section_size == 0:
-                        logger.debug(f"Skipping section {section_name} with zero size")
                         continue
 
-                    # Calculate adjusted section size after symbol subtraction
-                    unique_sec = f"{segment_name}.{section_name}"
-                    adjusted = section_size - section_subtractions.get(unique_sec, 0)
-                    if adjusted <= 0:
-                        logger.debug(
-                            f"Skipping section {unique_sec} - no remaining size {adjusted} after symbol subtraction"
+                    key = f"{segment_name}.{section_name}"
+                    subtraction = section_subtractions.get(key, 0)
+                    if subtraction > section_size:
+                        logger.warning(
+                            f"Section {key}: symbol bytes ({subtraction:,}) exceed section size ({section_size:,})."
                         )
+                        subtraction = section_size
```

Comment:
would a miscalculation situation like this ever warrant a sentry error for us? 

i feel like this sort of thing is more significant than a log warning , no?